src/main/java/com/flightstats/hub/webhook/LocalWebhookManager.java

```diff
@@ -22,38 +21,44 @@
 import java.util.function.Consumer;
 
 @Singleton
+@Slf4j
 public class LocalWebhookManager {
 
-    private static final Logger logger = LoggerFactory.getLogger(LocalWebhookManager.class);
+    private Map<String, WebhookLeader> localLeaders = new ConcurrentHashMap<>();
+
     private final KeyLockManager lockManager;
     @Inject
     @Named("Webhook")
     private Dao<Webhook> webhookDao;
     @Inject
     private Provider<WebhookLeader> v2Provider;
-    private Map<String, WebhookLeader> localLeaders = new ConcurrentHashMap<>();
+
+    private static int shutdownThreadCount;
 
     @Inject
-    public LocalWebhookManager() {
+    public LocalWebhookManager(@Named("Webhook") Dao<Webhook> webhookDao,
+                                   Provider<WebhookLeader> v2Provider,
+                               WebhookProperties webhookProperties) {
+        shutdownThreadCount = webhookProperties.getShutdownThreadCount();
```

Comment:
I don't think it's safe to set this static variable in the initializer. It's possible to call `runAndWait` prior to an instance being created, which would set the thread count to 0...unless it's bound as an eager singleton when the app's set up, which I'd rather not depend on.